sdk/common/src/main/java/io/opentelemetry/sdk/common/export/CompletableResultCode.java

```diff
@@ -107,4 +109,33 @@ public CompletableResultCode whenComplete(Runnable action) {
     }
     return this;
   }
+
+  /** Returns whether this {@link CompletableResultCode} has completed. */
+  public boolean isDone() {
+    synchronized (lock) {
+      return succeeded != null;
+    }
+  }
+
+  /** Calls {@link #fail()} if times out. */
+  public CompletableResultCode join(long timeout, TimeUnit unit) {
+    final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
If you don't mind copying in #1583 that'd be cool, I think the synchronous case optimization and less branching (same lines of code to just all `fail()` twice) is worth it. Then no worries closing that PR in favor of this one

Reply:
Will do.